tests/x11/firefox/firefox_downloading.pm

```diff
@@ -91,6 +91,8 @@ sub dl_menu {
     wait_still_screen 3,                                   6;
     send_key_until_needlematch 'firefox-downloading-menu', 'shift-f10', 3, 3;
     wait_still_screen 3,                                   6;
+    # sometimes menu does close due high load or some worker hickup, check & open menu again if not present
+    send_key_until_needlematch 'firefox-downloading-menu', 'shift-f10', 3, 3 unless check_screen('firefox-downloading-menu');
```

Comment:
`send_key_until_needlematch` already checks before sending keys, see https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1403 , so the additional `check_screen` is not necessary. Which means that you can use a loop:

```suggestion
    for (1 .. 2) {
        wait_still_screen 3,                                   6;
        send_key_until_needlematch 'firefox-downloading-menu', 'shift-f10', 3, 3;
    }
```

Reply:
done
https://openqa.suse.de/tests/7090721